### PR TITLE
Migrate: Service Id resource to Platform-go-SDK

### LINKED ISSUE #2560: Apply for iam_service_id fails with 404 when using IBMCLOUD_VISIBILITY=private
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Affected Resource(s)

* ibm_iam_service_id

### Terraform Configuration Files

Please include all Terraform configurations required to reproduce the bug. Bug reports without a functional reproduction may be closed without investigation.

```hcl
terraform {
  required_providers {
    ibm = {
      source  = "IBM-Cloud/ibm"
      version = ">= 1.23.2"
    }
  }
}

provider "ibm" {
}

resource "ibm_iam_service_id" "functional_user" {
  name = "test-private-functional-user"
}
```

### Expected Behavior

Apply should have worked

### Actual Behavior

```hcl
$ terraform apply -auto-approve
ibm_iam_service_id.functional_user: Creating...
╷
│ Error: Error creating serviceID: Request failed with status code: 404, ServerErrorResponse: <!DOCTYPE html>
│ <html>
│ <head>
│ <title>Error</title>
│ <style>
│     body {
│         width: 35em;
│         margin: 0 auto;
│         font-family: Tahoma, Verdana, Arial, sans-serif;
│     }
│ </style>
│ </head>
│ <body>
│ <h1>An 404 error occurred in iam-gateway.</h1>
│ </body>
│ </html>
│
│   with ibm_iam_service_id.functional_user,
│   on main.tf line 13, in resource "ibm_iam_service_id" "functional_user":
│   13: resource "ibm_iam_service_id" "functional_user" {
│
╵
```

### Steps to Reproduce

```
export IBMCLOUD_VISIBILITY=private
terraform init
terraform apply
```

Things work if `IBMCLOUD_VISIBILITY=public`

### PR DESCRIPTION
**Pull request by kavya498**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

```
=== RUN   TestAccIBMIAMServiceID_Basic
--- PASS: TestAccIBMIAMServiceID_Basic (61.18s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	64.073s
```
```
=== RUN   TestAccIBMIAMServiceID_import
--- PASS: TestAccIBMIAMServiceID_import (26.53s)
PASS
ok  	github.com/IBM-Cloud/terraform-provider-ibm/ibm	29.004s

```
Closes #2560

### CODE REVIEW

File: ibm/resource_ibm_iam_service_id.go [hunk 1]
```diff
@@ -60,88 +55,130 @@ func resourceIBMIAMServiceID() *schema.Resource {
...
 func resourceIBMIAMServiceIDCreate(d *schema.ResourceData, meta interface{}) error {
-	iamClient, err := meta.(ClientSession).IAMAPI()
+	iamIdentityClient, err := meta.(ClientSession).IAMIdentityV1API()
 	if err != nil {
 		return err
 	}
+
 	name := d.Get("name").(string)
 
 	userDetails, err := meta.(ClientSession).BluemixUserDetails()
```
Comment (hkantare) [inline ibm/resource_ibm_iam_service_id.go:80]: check for err != nil and return err

File: ibm/resource_ibm_iam_service_id.go [hunk 2]
```diff
@@ -60,88 +55,130 @@ func resourceIBMIAMServiceID() *schema.Resource {
 				Elem:     &schema.Schema{Type: schema.TypeString},
 				Set:      schema.HashString,
 			},
+			"unique_instance_crns": {
```
Comment (hkantare) [inline ibm/resource_ibm_iam_service_id.go:58]: Can you check a complete flow like create a service id and attach a service id policy to any service instance ...in the next run can you check will those values will be stored here (shd we make it as computed also)

File: ibm/resource_ibm_iam_service_id.go [hunk 3]
```diff
@@ -60,88 +55,130 @@ func resourceIBMIAMServiceID() *schema.Resource {
...
 	serviceIDUUID := d.Id()
-
-	serviceID, err := iamClient.ServiceIds().Get(serviceIDUUID)
+	getServiceIDOptions := iamidentityv1.GetServiceIDOptions{
+		ID: &serviceIDUUID,
+	}
+	serviceID, resp, err := iamIdentityClient.GetServiceID(&getServiceIDOptions)
 	if err != nil {
-		return fmt.Errorf("Error retrieving serviceID: %s", err)
+		if resp.StatusCode == 404 {
```
Comment (hkantare) [inline ibm/resource_ibm_iam_service_id.go:116]: check if resp != nil

File: ibm/resource_ibm_iam_service_id.go (hunk 1 above)
Comment (kavya498) [reply ibm/resource_ibm_iam_service_id.go:76]: done

**Pull request merged**